Windows plugin fails to compile on VS 18.6

<!-- **IMPORTANT: before filing a bug report, please make sure you checked that you have done everything that is covered in the readme as there are steps that need to be followed to get the plugin working correctly on both platforms. You can make use of the example app as a reference to see how the plugin behaves  -->

**Describe the bug**
After updating Visual Studio to 18.6 (released on May 12), projects that depend on this package fail to compile on Windows. The update to 18.6 updates the MSVC toolchain to 14.51 which [now errors on usage of experimental coroutine headers](https://github.com/microsoft/STL/pull/5804).

> C:\Program Files\Microsoft Visual Studio\18\Community\VC\Tools\MSVC\14.51.36231\include\experimental\coroutine(37,1): error C2338: static assertion failed: 'error STL1011: The /await compiler option, <experimental/coroutine>, <experimental/generator>, and <experimental/resumable> are deprecated by Microsoft and will be REMOVED SOON. They are superseded by the C++20 <coroutine> and C++23 <generator> headers. You can define _SILENCE_EXPERIMENTAL_COROUTINE_DEPRECATION_WARNINGS to suppress this error for now.' [D:\source\MyTests\flutter_notifications_cpp20_bug\build\windows\x64\plugins\flutter_local_notifications_windows\shared\flutter_local_notifications_windows.vcxproj]

I'm guessing this is coming from one of the dependencies, or a transitive dependency; At least I couldn't find any /await switch or \<experimental\> headers on the windows plugin code.

Forcing CPP20 generates another error (perhaps revealing a bug):

> <redacted>\windows\flutter\ephemeral\.plugin_symlinks\flutter_local_notifications_windows\src\plugin.cpp(36,33): error C2440: 'initializing': cannot convert from 'ATL::CW2A' to 'std::basic_string<char,std::char_traits<char>,std::allocator<char>>' [<redacted>\build\windows\x64\plugins\flutter_local_notifications_windows\shared\flutter_local_notifications_windows.vcxproj]

**To Reproduce**
1. Install VS 18.6
2. flutter create --platforms windows cpp20bug
3. flutter pub add flutter_local_notifications
4. flutter build windows

**Expected behavior**
It builds

--- a/.github/workflows/validate.yml
+++ b/.github/workflows/validate.yml
@@ -207,7 +207,7 @@ jobs:
           dart run msix:create
   build_example_windows_min_flutter:
     name: Build Windows example app (3.38.1)
-    runs-on: windows-latest
+    runs-on: windows-2022
     steps:
       - uses: actions/checkout@v6
       - uses: subosito/flutter-action@v2

--- a/flutter_local_notifications/CHANGELOG.md
+++ b/flutter_local_notifications/CHANGELOG.md
@@ -1,7 +1,7 @@
 ## [22.0.1]
 
 * [Windows] Suppress warning around usage of experimental coroutines. This is to fix issue [#2777](https://github.com/MaikuB/flutter_local_notifications/issues/2777)
-* 
+
 ## [22.0.0]
 
 * [Android] calling the `requestNotificationPolicyAccess()` method belonging to the `AndroidFlutterLocalNotificationsPlugin` class will now highlight associated application. Thanks to the PR from [Claudius Kienle](https://github.com/claudius-kienle)

--- a/flutter_local_notifications/CHANGELOG.md
+++ b/flutter_local_notifications/CHANGELOG.md
@@ -1,3 +1,7 @@
+## [22.0.1]
+
+* [Windows] Suppress warning around usage of experimental coroutines. This is to fix issue [#2777](https://github.com/MaikuB/flutter_local_notifications/issues/2777)
+* 
 ## [22.0.0]
 
 * [Android] calling the `requestNotificationPolicyAccess()` method belonging to the `AndroidFlutterLocalNotificationsPlugin` class will now highlight associated application. Thanks to the PR from [Claudius Kienle](https://github.com/claudius-kienle)

--- a/flutter_local_notifications/pubspec.yaml
+++ b/flutter_local_notifications/pubspec.yaml
@@ -2,7 +2,7 @@ name: flutter_local_notifications
 description: A cross platform plugin for displaying and scheduling local
   notifications for Flutter applications with the ability to customise for each
   platform.
-version: 22.0.0
+version: 22.0.1
 homepage: https://github.com/MaikuB/flutter_local_notifications/tree/master/flutter_local_notifications
 issue_tracker: https://github.com/MaikuB/flutter_local_notifications/issues
 
@@ -11,7 +11,7 @@ dependencies:
   flutter:
     sdk: flutter
   flutter_local_notifications_linux: ^8.0.1
-  flutter_local_notifications_windows: ^3.1.0
+  flutter_local_notifications_windows: ^3.1.1
   flutter_local_notifications_web: ^1.0.0
   flutter_local_notifications_platform_interface: ^12.0.0
   timezone: ^0.11.0

--- a/flutter_local_notifications_windows/CHANGELOG.md
+++ b/flutter_local_notifications_windows/CHANGELOG.md
@@ -1,3 +1,7 @@
+## [3.1.1]
+
+* Suppress warning around usage of experimental coroutines. This is to fix issue [#2777](https://github.com/MaikuB/flutter_local_notifications/issues/2777)
+
 ## [3.1.0]
 
 * Updated `periodicallyShow()` to accept `payload` and `notificationDetails` as optional parameters

--- a/flutter_local_notifications_windows/pubspec.yaml
+++ b/flutter_local_notifications_windows/pubspec.yaml
@@ -1,6 +1,6 @@
 name: flutter_local_notifications_windows
 description: Windows implementation of the flutter_local_notifications plugin
-version: 3.1.0
+version: 3.1.1
 homepage: https://github.com/MaikuB/flutter_local_notifications/tree/master/flutter_local_notifications_windows
 issue_tracker: https://github.com/MaikuB/flutter_local_notifications/issues
 

--- a/flutter_local_notifications_windows/src/CMakeLists.txt
+++ b/flutter_local_notifications_windows/src/CMakeLists.txt
@@ -5,6 +5,8 @@ cmake_minimum_required(VERSION 3.10)
 
 project(flutter_local_notifications_windows_library VERSION 1.0.0 LANGUAGES CXX)
 
+add_compile_definitions(_SILENCE_EXPERIMENTAL_COROUTINE_DEPRECATION_WARNINGS)
+
 add_library(flutter_local_notifications_windows SHARED
   "ffi_api.cpp"
   "plugin.cpp"

